Source/CombatExtended/CombatExtended/Loadouts/PawnColumnWorker_Loadout.cs

```diff
@@ -78,21 +79,43 @@ public override void DoCell(Rect rect, Pawn pawn, PawnTable table)
             float num3 = rect.x;
             //added:
             float num4 = rect.y + ((rect.height - IconSize) / 2);
+            float equipNowWidth = "CE_UpdateLoadoutNow".Translate().GetWidthCached();
 
             // Reduce width if we're adding a clear forced button
             bool somethingIsForced = pawn.HoldTrackerAnythingHeld();
-            Rect loadoutButtonRect = new Rect(num3, rect.y + 2f, (float)num, rect.height - 4f);
+            Rect loadoutRect = new Rect(num3, rect.y + 2f, (float)num, rect.height - 4f);
             if (somethingIsForced)
             {
-                loadoutButtonRect.width -= 4f + (float)num2;
+                loadoutRect.width -= 4f + (float)num2;
             }
 
             // Main loadout button
-            string label = pawn.GetLoadout().label.Truncate(loadoutButtonRect.width, null);
-            Widgets.Dropdown<Pawn, Loadout>(loadoutButtonRect, pawn, (Pawn p) => p.GetLoadout(), new Func<Pawn, IEnumerable<Widgets.DropdownMenuElement<Loadout>>>(Button_GenerateMenu), label, null, null, null, null, true);
+            Rect mainLoadoutButton = pawn.Spawned ? loadoutRect.LeftPartPixels(loadoutRect.width - equipNowWidth - 16) : loadoutRect;
+            mainLoadoutButton.xMax -= 2;
+            string label = pawn.GetLoadout().label.Truncate(mainLoadoutButton.width, null);
+            Widgets.Dropdown<Pawn, Loadout>(mainLoadoutButton, pawn, (Pawn p) => p.GetLoadout(), new Func<Pawn, IEnumerable<Widgets.DropdownMenuElement<Loadout>>>(Button_GenerateMenu), label, null, null, null, null, true);
+
+            if (pawn.Spawned)
+            {
+                Rect forceEquipNow = loadoutRect.RightPartPixels(equipNowWidth + 12);
+                if (Widgets.ButtonText(forceEquipNow, "CE_UpdateLoadoutNow".Translate()))
+                {
+                    Job job = JobGiver_UpdateLoadout.GetUpdateLoadoutJob(pawn);
+                    CompInventory inventory = pawn.TryGetComp<CompInventory>();
+                    if (inventory != null)
```

Comment:
Will `inventory` ever be null when `job` is non-null?  In other words, does `GetUpdateLoadoutJob` check for null inventory?  If that is the expectation, I don't mind the explicit null check, but it might be worth issuing a warning if it ever is the case that we're doing a loadout job with a null inventory.

Reply:
If it's a race that doesn't use base pawn. Better be safe imo.